### PR TITLE
Setting up Jest and Enzyme tests

### PR DESCRIPTION
**Pull request by crcommons**
This pull request makes the following changes:
- Adds Jest and Enzyme.
- Adds snapshot test for test component
- Adds expect test for test component

Testing checklist:
- [ ] Run npm run test; all tests should pass
- [ ] Closely review code

- [x] I certify that I ran my checklist

Fixes ushahidi/platform# .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by crcommons** [comment]
Ugh, I just realized that I should adjust the travis scripts to run these tests too. I will add in the morning.

File: .eslintrc.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     },
     "rules": {
         "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
-        "prettier/prettier": ["error", { "singleQuote": true }, {"trailingComma" :true}]
+        "comma-dangle": ["error", "never"],
+        "prettier/prettier": ["error", { "singleQuote": true }]
```
Comment (rjmackay) [inline .eslintrc.json]: Is prettier going to place nice with airbnb styles? For some reason I thought that clashed, hopefully I'm just wrong :)

File: package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "lint-fix": "eslint . --fix",
     "precommit": "lint-staged",
     "cz": "git-cz",
-    "retry": "git-cz --retry"
+    "retry": "git-cz --retry",
+    "test": "jest"
```
Comment (rjmackay) [inline package.json]: Ideally `npm run test` should run all tests - so jest and our old ugly tests.

File: package.json [hunk 3]
```diff
@@ -33,15 +34,23 @@
       "git add"
     ]
   },
+  "jest": {
+    "roots": [
+      "./tests/migration"
```
Comment (rjmackay) [inline package.json]: Could we do side-by-side tests instead? ie. put the tests in `app/` beside thing component we're testing, just named thing.spec.js or thing.test.js?
I think jests default `testMatch` setting works for this.

File: package.json [hunk 4]
```diff
@@ -33,15 +34,23 @@
...
+    ],
+    "setupTestFrameworkScriptFile": "./tests/setup.js"
```
Comment (rjmackay) [inline package.json]: Maybe name this jest-setup.js while we're stuck in transition mode?

File: .eslintrc.json [hunk 5]
```diff
@@ -19,6 +19,7 @@
     },
     "rules": {
         "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
-        "prettier/prettier": ["error", { "singleQuote": true }, {"trailingComma" :true}]
+        "comma-dangle": ["error", "never"],
```
Comment (rjmackay) [inline .eslintrc.json]: why the specific comma dangle rules?

File: gulpfile.babel.js [hunk 6]
```diff
@@ -126,7 +126,7 @@ gulp.task('clean', (done) => {
   });
 });
 
-
+// TO-DO: DECIDE IF WE WANT TO KEEP GULP TESTING
```
Comment (rjmackay) [inline gulpfile.babel.js]: Burn it 🔥

**Pull request merged**